src/components/BookPlace.tsx

```diff
@@ -133,23 +167,37 @@ const BookPlace: React.FC<BookPlaceProps> = ({ onClose, placeId, placeBookings,
           label='Forever booking'
         />
         <Button disabled={disabledFlag} onClick={onBookingDatesOpen}>
-          {bookingDateRange ?
-            `Booked Days: ${dateToString(bookingDateRange.startDate)} - 
-            ${bookingDateRange?.endDate ? dateToString(bookingDateRange.endDate) : 'Forever'}` :
-            'Choose Booking Dates'
+          {bookingDateRange
+            ? `Booked Days: ${dateToString(bookingDateRange.startDate)} - 
+            ${bookingDateRange?.endDate ? dateToString(bookingDateRange.endDate) : 'Forever'}`
+            : 'Choose Booking Dates'
           }
         </Button>
-        <InputGroup 
+        <select
+          value={visitorName}
           required
+          onChange={onSelectUserChange}
           disabled={disabledFlag}
-          placeholder='Name'
-          value={visitorName}
-          onChange={onNameChange}
-        />
+        >
+          <option value={placeholderValue} selected>Choose user</option>
+          {members.map(member =>
+            <option key={member.id} id={member.id} value={JSON.stringify({ name: member.data().name, id: member.id })}>
```

Comment:
If I remember that right, React allows JS object to be the value.
So there is no need to stringify it